### PR TITLE
Add after_success and branch examples (#1)

### PR DESCRIPTION
**Pull request by gravyboat**
* Add after_success and branch examples

* Remove a comment

### CODE REVIEW

**Comment by gravyboat** [comment]
This resolves https://github.com/travis-ci/travis-ci/issues/5065.

**Comment by gravyboat** [comment]
@plaindocs Is there anything missing from this that's keeping it from being merged? If so please let me know and I'll fix it up.

File: user/docker.md [hunk 1]
```diff
@@ -137,6 +137,20 @@ Within your `.travis.yml` prior to attempting a `docker push` or perhaps before
 docker login -e="$DOCKER_EMAIL" -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
 ```
 
+#### Branch Based Registry Pushes
+
+Using `after_success` it is also possible to push an image to a registry based
+on the branch:
+
```
Comment (plaindocs) [inline user/docker.md]: I'd rephrase this to something like:

    To push a particular branch of your repository to a remote registry, use the `after_success` section of your `.travis.yml`:

> you also don't need to indent fenced code blocks.

**Comment by plaindocs** [comment]
Comments above, sorry for the delay.

**Comment by gravyboat** [comment]
No worries, I'll update this as soon as I get a chance.

**Comment by gravyboat** [comment]
Okay this is all fixed up, thanks for the feedback.

**Comment by plaindocs** [comment]
Thanks! :-)

**Comment by Silex** [comment]
This fails for pull requests. You need to use something like `if [ "${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH}" == "master"]`
